src/main/java/org/opentripplanner/service/worldenvelope/service/WorldEnvelopeModel.java

```diff
@@ -0,0 +1,47 @@
+package org.opentripplanner.service.worldenvelope.service;
+
+import java.io.Serializable;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import org.opentripplanner.service.worldenvelope.model.WorldEnvelope;
+import org.opentripplanner.service.worldenvelope.model.WorldEnvelopeService;
+
+/**
+ * Provide a container for the envelope. This is used set and pass the immutable envelope into the
+ * application context.
+ * <p>
+ * The envelope is bound to the model at graph build time. Ba aware
+ * that, before that the envelope is not set, and accessing it will cause
+ * a null pointer exception.
+ * <p>
+ * This class serve both the role as model and service.
+ * <ol>
+ *   <li>As model it provide a place to store and update the envelope</li>
+ *   <li>
+ *     As service it provide read-only access the the envelope for APIs. It implements the
+ *     {@link WorldEnvelopeService} to provide this role.
+ *   </li>
+ * </ol>
+ * <p>
+ * THIS CLASS IS THREAD-SAFE.
+ */
+public class WorldEnvelopeModel implements Serializable, WorldEnvelopeService {
```

Comment:
I'd be tempted to have the service sub package at the root of "org.opentripplanner.service.worldenvelope" as it's already explicitly part of the service package.

Reply:
I would like to make the WorldEnvelope an example to follow, so lets discuss it next week. Usually the "aggregate root" is put in the root package: `o.o.s.worldenvelope.WorldenvElopeService`.

I want to move the `CalculateWorldEnvelopeModule` (in graph_builder) inside `o.o.s.worldenvelope.nnn` also, but we can not do that before `graph_builder` is refactored - I do not want circular dependencies inside the world envelope.